### PR TITLE
feat: add parsing of enum values

### LINKED ISSUE #104: Add parsing of enum args
Many CLI have arguments that takes one of given values.

Example:
```--format <checkstyle|compact|json|junit|markdown|unix>```

Please, add parsing of such enum args.

### PR DESCRIPTION
**Pull request by ifedchankau**
Add parsing of enum values of arguments

Closes #104

### CODE REVIEW

File: src/parseSection.js [hunk 1]
```diff
@@ -68,6 +69,22 @@ const splitSection = (section) => {
...
 };
 
+/**
+ * Set enum values for argument if it exist
+ * @param {object} argument - argument template
+ * @param {string} option - argument with description from Options section
+ * @param {object} context - internal config
+ * @return {object} argument - argument template with enum properties
+ */
+const setEnum = (argument, option, context) => {
```
Comment (romanrostislavovich) [inline src/parseSection.js]: Please, use pure function

**Review by romanrostislavovich — COMMENTED** [review]
Please, use pure function

File: src/parseSection.js [hunk 2]
```diff
@@ -68,6 +70,21 @@ const splitSection = (section) => {
...
+/**
+ * Searches for enum values of an argument
+ * @param {string} option - argument with description from Options section
+ * @param {object} context - internal config
+ * @return {array} enumValues - enum values of argument
+ */
+const getEnum = (option, context) => {
+    let enumValues = [];
+    const match = option.match(context.regexp.enumValues.enum);
+    if (match) {
```
Comment (romanrostislavovich) [inline src/parseSection.js]: Please, use ternary operator

File: src/parseSection.js [hunk 3]
```diff
@@ -8,6 +8,8 @@ const options = (section, context, argumentTemplate) => {
                 let argumentNames = '';
                 let argumentDescription = '';
 
+                const argumentEnumValues = getEnum(option, context);
+                if (argumentEnumValues[0]) argument.enum = argumentEnumValues;
```
Comment (romanrostislavovich) [inline src/parseSection.js]: Maybe exist another way to validate of empty array?

File: src/parseSection.js (hunk 1 above)
Comment (ifedchankau) [reply src/parseSection.js]: @svoboda-rabstvo done

File: src/parseSection.js (hunk 2 above)
Comment (ifedchankau) [reply src/parseSection.js]: @svoboda-rabstvo done

File: src/parseSection.js (hunk 3 above)
Comment (ifedchankau) [reply src/parseSection.js]: @svoboda-rabstvo yes, i have changed it.

**Review by romanrostislavovich — APPROVED** [review]
LGTM

File: src/parseSection.js [hunk 4]
```diff
@@ -68,6 +72,18 @@ const splitSection = (section) => {
...
+/**
+ * Searches for enum values of an argument
+ * @param {string} option - argument with description from Options section
+ * @param {object} context - internal config
+ * @return {array} - enum values of argument
+ */
+const getEnum = (option, context) => {
+    const match = option.match(context.regexp.enumValues.enum);
+    return match ?
+        match[1].split(context.regexp.enumValues.split) : [];
```
Comment (romanrostislavovich) [inline src/parseSection.js:84]: Please, check the `match` onto `null` or/and `empty` or it's don't need?

Comment (ifedchankau) [reply src/parseSection.js:84]: @svoboda-rabstvo, yes, it's not necessary in this function. It is necessary only on final function of parser, where template is filling. So, `templatizer` function should have (and has, just look in that PR) that check if enum empty despite of checking it in function `getEnum()`.

Comment (romanrostislavovich) [reply src/parseSection.js:84]: Ok

**Comment by ifedchankau** [comment]
PR was closed by a mistake. Moved to #130

**Pull request closed without merging**